geom/src/polygon.rs

```diff
@@ -216,28 +190,31 @@ impl Polygon {
     }
 
     /// Top-left at the origin. Doesn't take Distance, because this is usually pixels, actually.
-    pub fn rectangle(width: f64, height: f64) -> Polygon {
-        Polygon {
-            points: vec![
-                Pt2D::new(0.0, 0.0),
-                Pt2D::new(width, 0.0),
-                Pt2D::new(width, height),
-                Pt2D::new(0.0, height),
-                Pt2D::new(0.0, 0.0),
-            ],
-            indices: vec![0, 1, 2, 0, 2, 3],
-            rings: None,
-        }
+    pub fn maybe_rectangle(width: f64, height: f64) -> Result<Self> {
+        Ring::new(vec![
+            Pt2D::new(0.0, 0.0),
+            Pt2D::new(width, 0.0),
+            Pt2D::new(width, height),
+            Pt2D::new(0.0, height),
+            Pt2D::new(0.0, 0.0),
+        ])
+        .map(|ring| ring.into_polygon())
     }
 
-    pub fn rectangle_centered(center: Pt2D, width: Distance, height: Distance) -> Polygon {
-        Polygon::rectangle(width.inner_meters(), height.inner_meters()).translate(
+    /// Top-left at the origin. Doesn't take Distance, because this is usually pixels, actually.
+    /// Note this will panic if `width` or `height` is 0.
+    pub fn rectangle(width: f64, height: f64) -> Self {
```

Comment:
There are so many callers :(  I didn't audit them yet, meaning there could be some crashes. I'm willing to accept that risk for now.